Спринт ДЗ№23

CasTeTRU opened the pull request.

CasTeTRU committed:
Спринт 23
Добавлена иконка

CasTeTRU committed:
Спринт 23
Промежуточная версия

CasTeTRU committed:
Спринт 23
Финальная версия

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/playlist/data/impl/PlaylistRepositoryImpl.kt:
```diff
@@ -66,5 +67,65 @@ class PlaylistRepositoryImpl(
+    }
+
+    override suspend fun deletePlaylist(playlistId: Long) {
+        val playlist = getPlaylistById(playlistId) ?: throw IllegalStateException("Playlist with id $playlistId not found")
```
Тут не стоит эксепшн выбрасывать, если плейлист не найден, то можно просто ничего не удалять - плейлист и так удалён

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/playlist/ui/CreatePlaylistFragment.kt:
```diff
@@ -297,6 +359,50 @@ class CreatePlaylistFragment : Fragment() {
         snackbar.show()
     }
 
+    private fun setCursorColor(editText: android.widget.EditText) {
```
Так переопределять цвет курсора не стоит - использование рефлексии может привести к различным сайд эффектам и даже крешам. Лучше переопределить тему EditText и в этом теме переопределить цвет курсора - это рекомендованный путь, так как менять дровабл не стоит, так как сам курсор рисует система, а цвет выбирается из темы

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/playlist/ui/CreatePlaylistFragment.kt:
```diff
@@ -27,12 +28,15 @@ import org.koin.androidx.viewmodel.ext.android.viewModel
 import java.io.FileOutputStream
 
-class CreatePlaylistFragment : Fragment() {
+open class CreatePlaylistFragment : Fragment() {
```
Класс не стоит делать открытым - у него нет наследников

ArturNurtdinov reviewed (commented):
1. Экран плейлиста в светлой теме не соответствует макетам - нужно исправить цвет текста количества минут и количества треков, цвет текста описания, убрать закругление углов обложки, исправить высоту списка треков:

<img width="380" height="816" alt="image" src="https://github.com/user-attachments/assets/4660f878-f8cf-4187-9426-5ea3286f1cfc" />

Макет:

<img width="352" height="783" alt="image" src="https://github.com/user-attachments/assets/001ab5a7-47ae-43d4-890e-b74ca0f7dd6b" />

2. В тёмной теме на экране плейлиста над обложкой отображается тёмная полоска:

<img width="569" height="669" alt="image" src="https://github.com/user-attachments/assets/791482a5-1333-4da4-afa5-f5c7937cb1fd" />

CasTeTRU committed:
Спринт 23
Отработаны замечания:
1) Последние добавленные треки находятся в верхней части списка.
2) Экран плейлиста в светлой теме не соответствует макетам - нужно исправить цвет текста количества минут и количества треков и убрать закругление углов обложки, высоту списка треков
 3) В тёмной теме на экране плейлиста над обложкой отображается тёмная полоска

CasTeTRU committed:
Спринт 23
Минификс

CasTeTRU committed:
Спринт 23
Минификс 2.0

CasTeTRU committed:
Небольшие правки не в зачёт
